be/src/formats/parquet/meta_helper.cpp

```diff
@@ -53,51 +42,121 @@ void ParquetMetaHelper::build_column_name_2_pos_in_meta(
 }
 
 void ParquetMetaHelper::prepare_read_columns(const std::vector<HdfsScannerContext::ColumnInfo>& materialized_columns,
-                                             std::vector<GroupReaderParam::Column>& read_cols) const {
+                                             std::vector<GroupReaderParam::Column>& read_cols,
+                                             std::unordered_set<std::string>& existed_column_names) const {
     for (auto& materialized_column : materialized_columns) {
         int32_t field_idx = _file_metadata->schema().get_field_idx_by_column_name(materialized_column.name());
         if (field_idx < 0) continue;
 
-        auto parquet_type = _file_metadata->schema().get_stored_column_by_field_idx(field_idx)->physical_type;
+        const ParquetField* parquet_field = _file_metadata->schema().get_stored_column_by_field_idx(field_idx);
+        // check is type is invalid
+        if (!_is_valid_type(parquet_field, &materialized_column.slot_desc->type())) {
+            continue;
+        }
+
+        auto parquet_type = parquet_field->physical_type;
         GroupReaderParam::Column column = _build_column(field_idx, parquet_type, materialized_column.slot_desc,
                                                         materialized_column.decode_needed);
         read_cols.emplace_back(column);
+        existed_column_names.emplace(Utils::format_name(materialized_column.name(), _case_sensitive));
     }
 }
 
+bool ParquetMetaHelper::_is_valid_type(const ParquetField* parquet_field, const TypeDescriptor* type_descriptor) const {
+    if (type_descriptor->is_unknown_type()) {
+        return false;
+    }
+    // only check for complex type now
+    // if complex type has none valid subfield, we will treat this struct type as invalid type.
+    if (!parquet_field->type.is_complex_type()) {
+        return true;
+    }
+
+    if (parquet_field->type.type != type_descriptor->type) {
+        // complex type mismatched
+        return false;
+    }
+
+    bool has_valid_child = false;
+
+    if (parquet_field->type.is_array_type() || parquet_field->type.is_map_type()) {
+        for (size_t idx = 0; idx < parquet_field->children.size(); idx++) {
+            if (_is_valid_type(&parquet_field->children[idx], &type_descriptor->children[idx])) {
+                has_valid_child = true;
```

Comment:
for a map with scalar type key, we always get it valid, but may be we just access values.

Reply:
don't need to do it